packages/amplify_api/test/amplify_api_graphql_helpers_test.dart

```diff
@@ -46,4 +57,50 @@ void main() {
     }
     fail("Expected an ApiException");
   });
+
+  test('Query returns a GraphQLRequest<Blog> when provided a modelType',
+      () async {
+    String id = UUID.getUUID();
+    GraphQLRequest<Blog> req = ModelQueries.get<Blog>(Blog.classType, id);
+    List<GraphQLResponseError> errors = [];
+    String data = '''{
+        "getBlog": {
+            "createdAt": "2021-01-01T01:00:00.000000000Z",
+            "id": "$id",
+            "name": "TestAppBlog"
+        }
+    }''';
+
+    GraphQLResponse<Blog> response = GraphQLResponseDecoder.instance
+        .decode<Blog>(request: req, data: data, errors: errors);
+
+    expect(response.data.runtimeType, Blog);
+  });
+
+  test('Query returns a GraphQLRequest<String> when not provided a modelType',
+      () async {
+    String id = UUID.getUUID();
+    String doc = '''query MyQuery {
+      getBlog {
+        id
+        name
+        createdAt
+      }
+    }''';
+    GraphQLRequest<String> req =
+        GraphQLRequest(document: doc, variables: {id: id});
+    List<GraphQLResponseError> errors = [];
+    String data = '''{
+        "getBlog": {
+            "createdAt": "2021-01-01T01:00:00.000000000Z",
+            "id": "$id",
+            "name": "TestAppBlog"
+        }
+    }''';
+
+    GraphQLResponse<String> response = GraphQLResponseDecoder.instance
+        .decode<String>(request: req, data: data, errors: errors);
+
+    expect(response.data.runtimeType, String);
```

Comment:
Same as above. Better to compare the value than the type, since it's already guaranteed.